CRM/Contact/BAO/SavedSearch.php

```diff
@@ -153,12 +153,8 @@ public static function getFormValues($id) {
           }
         }
       }
-      if (substr($element, 0, 7) == 'custom_' &&
-        (substr($element, -5, 5) == '_from' || substr($element, -3, 3) == '_to')
-      ) {
-        // Ensure the _relative field is set if from or to are set to ensure custom date
-        // fields with 'from' or 'to' values are displayed when the are set in the smart group
-        // being loaded. (CRM-17116)
+      // We should only set the relative key for custom date fields if it is not already set in the array.
+      if (substr($element, 0, 7) == 'custom_' && CRM_Contact_BAO_Query::isCustomDateField($element)) {
```

Comment:
Is this right - ie we don't need to strip _from & _to from $element before passing to isCustomDateField?

Reply:
Yep you were right have fixed up